src/main/java/de/tum/in/www1/artemis/repository/StudentParticipationRepository.java

```diff
@@ -158,20 +158,23 @@ Optional<StudentParticipation> findWithEagerLegalSubmissionsByExerciseIdAndStude
             LEFT JOIN FETCH p.submissions s
             LEFT JOIN FETCH s.results r
             LEFT JOIN FETCH r.assessor
-            LEFT JOIN FETCH r.feedbacks
-            WHERE p.exercise.id = :#{#exerciseId}
-            AND p.testRun = :#{#testRun}
+            LEFT JOIN FETCH r.feedbacks f
+            LEFT JOIN FETCH f.testCase
+            WHERE p.exercise.id = :exerciseId
+            AND p.testRun = :testRun
             """)
     List<StudentParticipation> findByExerciseIdAndTestRunWithEagerSubmissionsResultAssessorFeedbacks(@Param("exerciseId") Long exerciseId, @Param("testRun") boolean testRun);
```

Comment:
Does the order matter here?
If not, a set might be more efficient. If the order mattters we should probably enforce the order using order by